extensions/binding/org.eclipse.smarthome.binding.sonyaudio/src/main/java/org/eclipse/smarthome/binding/sonyaudio/internal/protocol/SonyAudioConnection.java

```diff
@@ -0,0 +1,495 @@
+                }
+
+                SwitchNotifications switchNotifications = new SwitchNotifications(notifications.enabled,
+                        notifications.disabled);
+                audio_socket.callMethod(switchNotifications);
+            }
+
+            if (system_socket.getURI().equals(resource)) {
+                notifications = getSwitches(system_socket, notifications);
+
+                for (Iterator<Notification> iter = notifications.disabled.listIterator(); iter.hasNext();) {
+                    Notification a = iter.next();
+                    if (a.name.equalsIgnoreCase("notifyPowerStatus")) {
+                        notifications.enabled.add(a);
+                        iter.remove();
+                    }
+                }
+
+                SwitchNotifications switchNotifications = new SwitchNotifications(notifications.enabled,
+                        notifications.disabled);
+                system_socket.callMethod(switchNotifications);
+            }
+            listener.updateConnectionState(true);
+        } catch (IOException e) {
+            listener.updateConnectionState(false);
+        }
+    }
+
+    public synchronized void connect(ScheduledExecutorService scheduler) {
+        try {
+            base_uri = new URI(String.format("ws://%s:%d/%s", host, port, path)).normalize();
+
+            URI wsAvContentUri = base_uri.resolve(base_uri.getPath() + "/avContent").normalize();
+            av_content_socket = new SonyAudioClientSocket(this, wsAvContentUri, scheduler);
+            av_content_socket.open();
+
+            URI wsAudioUri = base_uri.resolve(base_uri.getPath() + "/audio").normalize();
+            audio_socket = new SonyAudioClientSocket(this, wsAudioUri, scheduler);
+            audio_socket.open();
+
+            URI wsSystemUri = base_uri.resolve(base_uri.getPath() + "/system").normalize();
+            system_socket = new SonyAudioClientSocket(this, wsSystemUri, scheduler);
+            system_socket.open();
+        } catch (Throwable t) {
+            logger.error("exception during connect avContent to {}:{}", host, port, t);
+        }
+    }
+
+    public synchronized void close() {
+        if(av_content_socket != null) {
+          av_content_socket.close();
+        }
+        av_content_socket = null;
+
+        if(audio_socket != null) {
+          audio_socket.close();
+        }
+        audio_socket = null;
+
+        if(system_socket != null) {
+          system_socket.close();
+        }
+        system_socket = null;
+    }
+
+    private boolean checkConnection(SonyAudioClientSocket socket) {
+        if (!socket.isConnected()) {
+            logger.debug("checkConnection: try to connect to {}", socket.getURI().toString());
+            try {
+                socket.open();
+                return socket.isConnected();
+            } catch (Throwable t) {
+                logger.error("exception during connect to {}", socket.getURI().toString(), t);
```

Comment:
do not log errors unless it's a software bug and remove `toString()` from parameters.